NachoSpigot-Server/src/main/java/net/minecraft/server/PlayerList.java

```diff
@@ -574,6 +575,9 @@ public EntityPlayer moveToWorld(EntityPlayer entityplayer, int i, boolean flag,
             }
 
             Player respawnPlayer = cserver.getPlayer(entityplayer1);
+
+            Nacho.get().getLagCompensator().registerMovement(respawnPlayer, location);
```

Comment:
On respawn, the players history should be cleared, as that's dead data at this point